args/core/containers/hashed_sparse_set.hpp

```diff
@@ -0,0 +1,345 @@
+#pragma once
+#include <vector>
+#include <unordered_map>
+#include <type_traits>
+#include <algorithm>
+#include <stdexcept>
+#include <core/platform/platform.hpp>
+#include <core/types/primitives.hpp>
+
+/**
+ * @file hashed_sparse_set.hpp
+ */
+
+namespace args::core
+{
+	/**@class hashed_sparse_set
+	 * @brief Quick lookup contiguous map. The map is based on the concept of a sparse set and thus inherits it's lookup complexity and contiguous nature.
+	 * @tparam value_type The type to be used as the value.
+	 * @tparam dense_type Container to be used to store the values.
+	 * @tparam sparse_type Container to be used to store the keys.
+	 * @note With default container parameters iterators may be invalidated upon resize. See reference of std::vector.
+	 * @note Removing item might invalidate the iterator of the last item in the dense container.
+	 */
+	template <typename value_type, template<typename...> typename dense_type = std::vector, template<typename...> typename sparse_type = std::unordered_map>
+	class hashed_sparse_set
+	{
+	public:
+		using self_type = hashed_sparse_set<value_type, dense_type, sparse_type>;
+		using self_reference = self_type&;
+		using self_const_reference = const self_type&;
+
+		using value_reference = value_type&;
+		using value_const_reference = const value_type&;
+		using value_pointer = value_type*;
+
+		using sparse_container = sparse_type<value_type, size_type>;
+		using dense_container = dense_type<value_type>;
+
+		using iterator = typename dense_container::iterator;
+		using const_iterator = typename dense_container::const_iterator;
+
+	private:
+		dense_container m_dense;
+		sparse_container m_sparse;
+
+		size_type m_size = 0;
+		size_type m_capacity = 0;
+
+	public:
+		A_NODISCARD dense_container& dense() { return m_dense; }
+		A_NODISCARD const dense_container& dense() const { return m_dense; }
+
+		A_NODISCARD iterator begin() { return m_dense.begin(); }
```

Comment:
to adhere to the standard you should think about also implementing `cbegin()` and `cend()`next to `begin() const` and `end() const`